vendor/src/github.com/docker/libnetwork/drivers/bridge/setup_device.go

```diff
@@ -27,10 +29,15 @@ func setupDevice(config *NetworkConfiguration, i *bridgeInterface) error {
 	// was not supported before that.
 	kv, err := kernel.GetKernelVersion()
 	if err == nil && (kv.Kernel >= 3 && kv.Major >= 3) {
+		i.Link.Attrs().HardwareAddr = netutils.GenerateRandomMAC()
+		log.Debugf("Setting bridge mac address to %s", i.Link.Attrs().HardwareAddr)
```

Comment:
I assume it's there in case the netlink method works. This should be overwritten by CreateBridge if netlink fails and setMac is true. I'm sure this could be a lot cleaner, but I'm interested in maintaining a 1:1 relationship to the original patch which was proven to work.